hack/generator/pkg/jsonast/jsonast.go

```diff
@@ -496,8 +496,64 @@ func oneOfHandler(ctx context.Context, scanner *SchemaScanner, schema *gojsonsch
 		return results[0], nil
 	}
 
-	// bail out, can't handle this yet:
-	return astmodel.AnyType, nil
+	// If there's more than one option, synthesize a type.
+	// Note that this is required because Kubernetes CRDs do not support OneOf the same way
+	// OpenAPI does, see https://github.com/Azure/k8s-infra/issues/71
+	var fields []*astmodel.FieldDefinition
+
+	for i, t := range results {
+		// TODO: Consider deduping these, as there seem to be some examples where a OneOf
+		// TODO: has many references to the same type (either as Refs or unnamed). See:
+		// TODO: https://schema.management.azure.com/schemas/2015-04-08/Microsoft.DocumentDB.json
+		// TODO: databaseAccounts_apis_databases_settings_childResource for example
+		switch concreteType := t.(type) {
+		case *astmodel.DefinitionName:
+			// Just a sanity check that we've already scanned this definition
+			// TODO: Could remove this?
+			if _, ok := scanner.FindDefinition(*concreteType); !ok {
+				return nil, fmt.Errorf("couldn't find struct for definition: %v", concreteType)
+			}
+			fieldName := scanner.idFactory.CreateFieldName(concreteType.Name())
+
+			// TODO: JSON name is unimportant here because we will implement the JSON marshaller anyway
+			jsonName := scanner.idFactory.CreatePrivateIdentifier(concreteType.Name())
+			field := astmodel.NewFieldDefinition(fieldName, jsonName, concreteType).MakeOptional()
+			fields = append(fields, field)
+		case *astmodel.EnumType:
+			// Just a sanity check that we've already scanned this definition
+			fieldName := scanner.idFactory.CreateFieldName(concreteType.Name())
+
+			// TODO: JSON name is unimportant here because we will implement the JSON marshaller anyway
+			jsonName := scanner.idFactory.CreatePrivateIdentifier(concreteType.Name())
+			field := astmodel.NewFieldDefinition(fieldName, jsonName, concreteType).MakeOptional()
+			fields = append(fields, field)
+		case *astmodel.StructType:
+			// TODO: This name sucks but what alternative do we have?
+			name := fmt.Sprintf("object%v", i)
+			fieldName := scanner.idFactory.CreateFieldName(name)
+
+			// TODO: JSON name is unimportant here because we will implement the JSON marshaller anyway
+			jsonName := scanner.idFactory.CreatePrivateIdentifier(name)
+			field := astmodel.NewFieldDefinition(fieldName, jsonName, concreteType).MakeOptional()
+			fields = append(fields, field)
+		case *astmodel.PrimitiveType:
+			// TODO: This name sucks but what alternative do we have?
+			name := fmt.Sprintf("primitive%v", i)
```

Comment:
Can you use the name of the primitive type to make the name less sucky - `string4` `int3` etc?

Reply:
Yeah, can do this. None of the options are really mind-bogglingly good, but I've added this at least until we come up with something better.